### PR TITLE
[verible] Update verible version

### PR DESCRIPTION
**Pull request by msfschaffner**
This updates Verible, since the version we are using is pretty old (from last May).
This version has many improvements, including new features such as [LSP support for editors](https://github.com/chipsalliance/verible/tree/master/verilog/tools/ls).

Note: this will not be merged until an announcement has been sent out via email to give people time to update their local installs.

Signed-off-by: Michael Schaffner <msf@google.com>

### CODE REVIEW

**Review by tjaychen — APPROVED** [review]

**Comment by msfschaffner** [comment]
CC @hzeller @mithro

**Comment by hzeller** [comment]
LGTM :)

**Comment by msfschaffner** [comment]
This is pending a vendor in from https://github.com/lowRISC/ibex/pull/1621.

**Comment by msfschaffner** [comment]
Rebased and re-vendored Ibex.

**Comment by msfschaffner** [comment]
Vendoring in Ibex in this PR does not work since the fetch enable expects a different MUBI encoding. So, we have to wait until Ibex is merged together with the MUBI encoding changes: https://github.com/lowRISC/opentitan/pull/12153.

**Review by eunchan — APPROVED** [review]
LGTM. But LSP does not work unfortunately (created issue https://github.com/chipsalliance/verible/issues/1320 )

**Pull request merged**
